sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java

```diff
@@ -116,16 +117,53 @@ public Schema getSchema() {
 
   @Override
   public void verifyDeterministic()
-      throws org.apache.beam.sdk.coders.Coder.NonDeterministicException {}
+      throws org.apache.beam.sdk.coders.Coder.NonDeterministicException {
+    verifyDeterministic(schema);
+  }
+
+  private void verifyDeterministic(Schema schema)
+      throws org.apache.beam.sdk.coders.Coder.NonDeterministicException {
+    for (Field field : schema.getFields()) {
+      verifyDeterministic(field.getType());
+    }
+  }
+
+  private void verifyDeterministic(FieldType fieldType)
+      throws org.apache.beam.sdk.coders.Coder.NonDeterministicException {
+    switch (fieldType.getTypeName()) {
+      case MAP:
+        throw new NonDeterministicException(this, "Map-valued fields cannot be used in keys");
```

Comment:
Why is this? Can you explain more in the message?

Reply:
Standard issue in Beam - keys need to have a deterministic encoding, and maps are not guaranteed to be deterministically encoded. Added comment.